addons/website_sale_autocomplete/tests/test_ui.py

```diff
@@ -0,0 +1,42 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import patch, HttpCase, tagged
+from unittest.mock import Mock
+
+CONTROLLER_PATH = 'odoo.addons.website_sale_autocomplete.controllers.main.AutoCompleteController'
+MOCK_GOOGLE_ID = 'aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g/dj1kUXc0dzlXZ1hjUQ=='
+MOCK_API_KEY = 'Tm9ib2R5IGV4cGVjdHMgdGhlIFNwYW5pc2ggaW5xdWlzaXRpb24gIQ=='
+
+@tagged('post_install', '-at_install')
+class TestUI(HttpCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product = cls.env['product.product'].create({
+            'name': 'A test product',
+            'website_published': True,
+            'list_price': 1
+        })
+
+    def test_autocomplete(self):
+        with patch(f'{CONTROLLER_PATH}._perform_complete_place_search',
+                   new=Mock(return_value={
```

Comment:
That seems extremely complicated if you just want to replace a method with one returning a pre-set value.

Would likely be simpler to use `self.patch(AutoCompleteController, '_perform_complete_place_search', lambda self: ...)`.